app/services/LoginService.php

```diff
@@ -2,26 +2,92 @@
 
 namespace App\Services;
 
+use App\Logs\LoggingAuth;
+use App\Logs\LoggingFiles;
 use App\Models\LoginModel;
 use App\Models\Repositories\LoginRepository;
 
 class LoginService
 {
     public function __construct(
-        private readonly LoginRepository $repository
+        private readonly LoginModel $user,
+        private readonly LoginRepository $repository,
+        private readonly FileUpload $fileUploader,
+        private readonly LoggingFiles $fileLog,
+        private readonly LoggingAuth $authLog
     ) {
     }
 
+    public function loginAttempts($errors, $post)
+    {
+        if (isset($errors)) {
+            $ip = $_SERVER['REMOTE_ADDR'];
+            $attempts = 1;
+            $time = date('Y-m-d H:i:s');
+            $endTime = date('Y-m-d H:i:s', strtotime('+15 minutes'));
+            $data = $this->repository->getLoginAttemptsData($ip);
+            if (!isset($data['userIP'])) {
+                $this->repository->setLoginAttemptsData($ip, $attempts);
+            } else {
+                $this->repository->updateLoginAttempts($data['attempts'] + 1, $ip);
+            }
+            if (isset($data['attempts']) && $data['attempts'] >= 2) {
+                $this->repository->updateBlockTime($time, $ip);
+                $this->authLog->error(
+                    '',
+                    [
+                        'ip' => $ip,
+                        'email' => $post['email'],
+                        'blockStart' => $time,
+                        'blockEnd' => $endTime,
+                        'path' => 'authLogs.log',
+                    ]
+                );
+
+                return;
+            }
+        }
+    }
+
+    public function checkBlockTime()
+    {
+        $blocked = false;
+        $ip = $_SERVER['REMOTE_ADDR'];
+        $data = $this->repository->getLoginAttemptsData($ip);
+        if (isset($data['blockTime'])) {
+            $blockTime = $data['blockTime'];
+            $diff = ((time() - strtotime($blockTime)) / 60);
+            if ($diff >= 15) {
```

Comment:
MUST: 15 is the magic digit. This value has a hard link with  string '+15 minutes'